### PR TITLE
Sean arms post

### PR DESCRIPTION
**Pull request by oapostrophe**
Adds a post on my spring 2021 work and optimized related images.

For images, I initially used the path format I saw in other posts "/assets/...".  However I found that the images weren't showing up in the preview, so I changed them to "../assets/..." which works in the preview.  If this is incorrect for the actual site, let me know and I can change them back.

### CODE REVIEW

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title:  "Report: ARMS - reducing robot simulation code by 92%"
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:3]: Please use title case.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+author: "Sean O'Connor"
+categories: [ "robotics", "simulation", "gazebo", "sdf", "javascript"  ]
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:5]: These should be "tags" not "categories". See: https://raw.githubusercontent.com/compusciencing/compusciencing.github.io/master/post-template.md

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+image: assets/images/arms1/robot_image.png
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:6]: Remove this line.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+---
+*A concise and intuitive alternative for writing verbose robotics simulation languages, built with browser JavaScript.  For more, check out my [blog](https://oapostrophe.github.io/) and [GitHub](https:/github.com/oapostrophe).*
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:8]: Add a blank line between the front matter and this subtitle.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+## Project Background
+
+Let's say you're designing a robot.  This robot will be used to go over rough terrain as part of post-disaster search-and-rescue missions, so it's important that it be both tall enough to clear uneven ground and stable enough not to topple over on bumps.  You want to know: what height and width will offer the optimal combination of ground clearance and stability?
+
+One possible way to find out might be to build several different robots of varying dimensions and test how they perform.  But building each physical robot is expensive and time-consuming, not to mention that there might be a lot of feasible height-weight combinations.  So you also want to know: is there a way we can test different builds, without the cost of physically building every configuration we want to test?
+
+![Selife taken by a robot from the ARCs lab during its construction](../assets/2021-05-22-report-arms-spring2021/robot_selfie.png)
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:16]: As you mentioned, I would prefer the image path to be `/assests/...`

How did you test it out? It probably won't work if you just open the markdown file in your browser without using a local server.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+![plain cube](../assets/2021-05-22-report-arms-spring2021/gazebo_cube.png)
+
+*A gray cube on a gray background in Gazebo - so basic, you almost can't see it.*
+
+Below is the SDF code required to create this basic object:
+
+![1x1x1 cube in SDF](../assets/2021-05-22-report-arms-spring2021/sdf_cube_snippet.png)
+
+Look at all those tags!  I'm getting a headache already.  Wouldn't it be nice if instead we could write something like this:
+```
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:31]: Add a space before the back-ticks and the paragraph above. Also, give the style as text.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+*A concise and intuitive alternative for writing verbose robotics simulation languages, built with browser JavaScript.  For more, check out my [blog](https://oapostrophe.github.io/) and [GitHub](https:/github.com/oapostrophe).*
+
+## Project Background
+
+Let's say you're designing a robot.  This robot will be used to go over rough terrain as part of post-disaster search-and-rescue missions, so it's important that it be both tall enough to clear uneven ground and stable enough not to topple over on bumps.  You want to know: what height and width will offer the optimal combination of ground clearance and stability?
+
+One possible way to find out might be to build several different robots of varying dimensions and test how they perform.  But building each physical robot is expensive and time-consuming, not to mention that there might be a lot of feasible height-weight combinations.  So you also want to know: is there a way we can test different builds, without the cost of physically building every configuration we want to test?
+
+![Selife taken by a robot from the ARCs lab during its construction](../assets/2021-05-22-report-arms-spring2021/robot_selfie.png)  
+
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:17]: Remove this newline.

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+![ARMS graph](../assets/2021-05-22-report-arms-spring2021/arms_fig1.png)
+
+*Graph of the % of code saved based on input size*
+
+![ARMS graph 2](../assets/2021-05-22-report-arms-spring2021/arms_fig2.png)
+
+*The savings are even higher when considered by character count.*
+
+The source code for the app can be found [on my GitHub](https://github.com/swow2015/arms2).  The app can be quickly demoed by downloading ["dist/main.js"](https://github.com/oapostrophe/arms2/blob/main/dist/main.js), ["dist/app.html"](https://github.com/oapostrophe/arms2/edit/main/dist/app.html), and ["dist/style.css"](https://github.com/oapostrophe/arms2/blob/main/dist/style.css).  With these three in the same directory, you can simply open the HTML file and the app will run.
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:90]: Will you eventually push the code to https://github.com/ARMS/arms.github.io?

File: _posts/2021-05-22-report-arms-spring2021.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+
+The app currently supports the following (see demo above for sample usage):
```
Comment (anthonyjclark) [inline _posts/2021-05-22-report-arms-spring2021.md:92]: Add a newline after this line.

**Review by anthonyjclark — CHANGES_REQUESTED** [review]
Very nice write-up! Thanks Sean.

Just a few changes.

File: _posts/2021-05-22-report-arms-spring2021.md (hunk 6 above)
Comment (oapostrophe) [reply _posts/2021-05-22-report-arms-spring2021.md:31]: What do you mean by giving the style as text, like this?

```text
(code here)
```

File: _posts/2021-05-22-report-arms-spring2021.md (hunk 8 above)
Comment (oapostrophe) [reply _posts/2021-05-22-report-arms-spring2021.md:90]: I can if desired, previously I remember we had decided to make a fresh repository for the new code.

File: _posts/2021-05-22-report-arms-spring2021.md (hunk 5 above)
Comment (oapostrophe) [reply _posts/2021-05-22-report-arms-spring2021.md]: I just had GitHub generate a pages website based on the branch in my fork.  I also tried looking at the local Markdown preview.  I changed the paths to be /assets/... for now.

File: _posts/2021-05-22-report-arms-spring2021.md (hunk 8 above)
Comment (anthonyjclark) [reply _posts/2021-05-22-report-arms-spring2021.md:91]: I happen to like that URL, but I'm open to suggestions.

Comment (oapostrophe) [reply _posts/2021-05-22-report-arms-spring2021.md:91]: I'll add the new code and give it a push this evening!

**Review by anthonyjclark — APPROVED** [review]

**Pull request merged**